pkg/discovery/discover_test.go

```diff
@@ -0,0 +1,548 @@
+	a := getAgent()
+
+	tests := []struct {
+		name      string
+		asset     *anypoint.ExchangeAsset
+		exchgfile *anypoint.ExchangeFile
+		err       error
+	}{
+		{
+			name: "Should return nil and no error if the Exchange asset has no files",
+			asset: &anypoint.ExchangeAsset{
+				Name:  "Sample exchange asset 1",
+				Files: nil,
+			},
+			exchgfile: nil,
+			err:       nil,
+		},
+		{
+			name: "Should return nil and no error if the Exchange asset has a file that is not of expected classifier",
+			asset: &anypoint.ExchangeAsset{
+				Name: "Sample exchange asset 2",
+				Files: []anypoint.ExchangeFile{{
+					Classifier: "oas3",
+				},
+				},
+			},
+			exchgfile: nil,
+			err:       nil,
+		},
+		{
+			name: "Should return an exchange asset if it has a file that is of expected classifier",
+			asset: &anypoint.ExchangeAsset{
+				Name: "Sample exchange asset 3",
+				Files: []anypoint.ExchangeFile{{
+					Classifier: "oas",
+				},
+				},
+			},
+			exchgfile: &anypoint.ExchangeFile{
+				Classifier: "oas",
+			},
+			err: nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			sd, err := a.getExchangeAssetSpecFile(tc.asset)
+			assert.Equal(t, tc.exchgfile, sd)
+			assert.Equal(t, tc.err, err)
+		})
+	}
+}
+
+func TestSetOAS2Endpoint(t *testing.T) {
+	a := getAgent()
+
+	tests := []struct {
+		name        string
+		endPointURL string
+		specContent []byte
+		result      []byte
+		err         error
+	}{
+		{
+			name:        "Should return error if Endpoint URL is not valid",
+			endPointURL: "postgres://user:abc{def=ghi@sdf.com:5432",
+			specContent: []byte("{\"basePath\":\"google.com\",\"host\":\"\",\"schemes\":[\"\"],\"swagger\":\"2.0\"}"),
+			result:      []byte("{\"basePath\":\"google.com\",\"host\":\"\",\"schemes\":[\"\"],\"swagger\":\"2.0\"}"),
+			err: &url.Error{
+				Op:  "parse",
+				URL: "postgres://user:abc{def=ghi@sdf.com:5432",
+				Err: fmt.Errorf("net/url: invalid userinfo"),
+			},
+		},
+		{
+			name:        "Should return error if the spec content is not a valid JSON",
+			endPointURL: "http://google.com",
+			specContent: []byte("google.com"),
+			result:      []byte("google.com"),
+			err:         fmt.Errorf("invalid character 'g' looking for beginning of value"),
+		},
+		{
+			name:        "Should return spec that has OAS2 endpoint set",
+			endPointURL: "http://google.com",
+			specContent: []byte("{\"basePath\":\"google.com\",\"host\":\"\",\"schemes\":[\"\"],\"swagger\":\"2.0\"}"),
+			result:      []byte("{\"basePath\":\"\",\"host\":\"google.com\",\"schemes\":[\"http\"],\"swagger\":\"2.0\"}"),
+			err:         nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			spec, err := a.setOAS2Endpoint(tc.endPointURL, tc.specContent)
+
+			if err != nil {
+				assert.Equal(t, tc.err.Error(), err.Error())
+			}
+
+			assert.Equal(t, tc.result, spec)
+		})
+	}
+}
+
+func TestGetAuthPolicy(t *testing.T) {
```

Comment:
What happens in this test if the policies array has more than one item?

Reply:
I had added this test for temporary sake, but will get rid of it for now and we can add tests once @Atraya21 refactors it to support more types of policies